docs/modules/maintain-cluster/pages/enterprise-rest-api.adoc

```diff
@@ -0,0 +1,460 @@
+    </rest>
+</hazelcast>
+----
+--
+
+YAML::
++
+[source,yaml]
+----
+hazelcast:
+  rest:
+    enabled: true
+    token-validity-seconds: 300
+----
+====
+
+
+[#obtaining-a-token]
+== Obtaining a Token
+
+To obtain a token, you should send a `POST` request to the token endpoint at `/hazelcast/rest/api/v1/token`. The request should include the `username` and `password` in JSON format. If authentication is successful, the response will contain a valid token for the expiry duration.
+
+Example request using cURL:
+
+[source,shell]
+----
+curl -X 'POST' \
+  'http://localhost:8443/hazelcast/rest/api/v1/token' \
+  -H 'Content-Type: application/json' \
+  -d '{
+  "username": "restuser",
+  "password": "restpassword"
+}'
+----
+
+It returns the following response if successful:
+
+[source,json]
+----
+{
+  "token": "<JWT Token>"
+}
+----
+
+
+== Accessing the Swagger UI
+
+Hazelcast REST API's Swagger UI provides information about each endpoint, including required parameters, request and response structures and types, potential response codes, and example responses. You can use this UI to easily navigate and test different API calls directly from the interface. To access it:
+
+. Enable the REST API.
+. Start a Hazelcast member.
+. Go to `http://<host>:<port>/swagger-ui/index.html` where <host> and <port> are the running member's IP address/hostname and port, respectively.
+
+The following is an example view:
+
+On the Swagger page, each endpoint is listed with a caret icon on the right side as shown below.
+
+image::rest-api-swagger-listed-endpoints.png[]
+
+Click the caret icon to expand the details for the endpoint. For this example, let's expand the `GET /hazelcast/rest/api/v1/cluster` endpoint.
+
+image::rest-api-swagger-expanding-an-endpoint.png[]
+
+After expanding the endpoint, let's proceed to make a request. Click the `Try it out` button, enter the parameters (if any). There are no parameters for this example.
+
+image::rest-api-swagger-clicking-execute-button.png[]
+
+Click the `Execute` button to send the request. You will see the response under "Server response" section as shown below. You can also view the executed Curl command in the "Curl" section.
+
+image::rest-api-swagger-clicking-try-it-out-button.png[]
+
+
+== GET/POST/DELETE HTTP Request Examples
+
+All the requests in the REST API can return one of the following response types.
+
+* Successful Void/Boolean response which does not have body.
+
+* Successful response which returns data in JSON format. For example:
+
+[source,json]
+----
+{
+  "nodeState": "ACTIVE",
+  "clusterState": "ACTIVE",
+  "numberOfMembers": 1
+}
+----
+
+* Error response in JSON format. For example:
+
+[source,json]
+----
+{
+  "statusCode": 400,
+  "message": "Please provide a valid value."
+}
+----
+
+=== Retrieving Cluster Status
+
+You can retrieve the status of your cluster by making a GET request to endpoint at `/hazelcast/rest/api/v1/cluster/state`.
```

Comment:
Shouldn't it be `.../cluster/status` or just `.../cluster`?

and let's reword it as follows:

You can send a `GET` request to the endpoint at `/hazelcast/rest/api/v1/cluster/status` to retrieve the status of your cluster.

Reply:
We defined the `/hazelcast/rest/api/v1/cluster` endpoint to get the cluster status, the same as in the legacy REST API. Not sure if we should change it. Any comments @ihsandemir @emreyigit ?